### ISSUE TITLE
Blocking: When manually the application is launched it reads a xml file to load a dynamically generated Combo box,but when test stack white is launching the application the combo box is not getting loaded.why such thing is happening ??

### ISSUE THREAD

**Issue by soumenhrithik** [TestStack/White#430]
Object reference error is coming in applications log

**Comment by dpisanu** [comment]
How are you starting the application via White? Are you possibly missing
some arguments on process startup?

**Comment by soumenhrithik** [comment]
@dpisanu 

please find my code below..suggest  how should i launch the application so that it gets access of all related files ,manually the application is working great ... every folder and dll has all the permissions granted. The combo box control appears after login...

[TestFixture]
        class AppPocUsingWhite
        {

            // source exe file path. here it is application exe path
            private const string ExeSourceFile = @"C:\Program Files\Folder1\App.exe";
            
             //Global Variable to for Application launch
            private static TestStack.White.Application _application;
            
            
        //Global variable to get the Main window of application.
            private static TestStack.White.UIItems.WindowItems.Window _mainWindow;
          
         
           
            
            
            String ErrDesc = "";
            String TCName = "";
            String TCID = "";
            String ExecutionStatus = "";
            static int srlno = 0;
            String Batchname = "App_GUI";
            
            [NUnit.Framework.TestFixtureSetUp]
            public static void Init()
            {
               
                //start process for the above exe file location
                var psi = new System.Diagnostics.ProcessStartInfo(ExeSourceFile);
                // launch the process through white application
                _application = TestStack.White.Application.AttachOrLaunch(psi);
                System.Threading.Thread.Sleep(10000);
                _mainWindow = _application.GetWindow(SearchCriteria.ByText("App1"), InitializeOption.NoCache);
                
            }
}

**Comment by dpisanu** [comment]
So you are starting the Executable without any arguments.
Do you maybe require additional arguments?
How is the application normally started?

**Comment by soumenhrithik** [comment]
normally a simple double click ...

can you write a syntax how should i launch the app...

**Comment by dpisanu** [comment]
Double Click on the Executable or on some Shortcut on your Desktop?
If it is a Shortcut then check the properties. Sometimes there are extra arguments embedded in there which you will then also need to pass from your Test Code.

**Comment by soumenhrithik** [comment]
no shortcut , double clicking from actual physical path..can you give me the syntax how to start test if a window is opened without launching the app.. i think if i am able to do it then i can proceed further ...

**Comment by dpisanu** [comment]
If there are no extra arguments to be passed, then I suggest starting the process via System.Diagnostics.Process.

```javascript
var process = new Process { StartInfo = new ProcessStartInfo { FileName = ExeSourceFile } };
process.Start();
var automationProcess = TestStack.White.Application.Attach(process);
```

**Comment by soumenhrithik** [comment]
The code didn't work !!

**Comment by dpisanu** [comment]
Did not work in what regards?
Process was not started? White did not attach? Problem persists?

**Comment by soumenhrithik** [comment]
Yup , process started but after login same problem ,combo box was not there ,object reference error ..in app log

**Comment by dpisanu** [comment]
Then it is some issue with your Application.
Has nothing to do with White at all.

You just started your Application with standard .NET Tools for starting processes.
There must be some specialty to your Application that it requires something additionally.

Is there any way you can debug your Application code to see what the difference is at start up?

**Comment by soumenhrithik** [comment]
Finally i am able to do this ... I have launched the application manually and attach its process id in white .now the tests are running fine. I know its a workaround but i have collected the process id through code. so no issues are here. :) Thanks for your assistance.

**Comment by ybellones** [comment]
I had similar issue and similar workaround just to resolve this issue. Not a great workaround but currently it is the only workaround I have. :)

**Comment by maxinfet** [comment]
@soumenhrithik  and @ybellones so attaching to the process you launched with the process object itself didnt work but attaching using the process id did?

**Comment by soumenhrithik** [comment]
yes it worked by attaching process id...

**Comment by dholgaurav** [comment]
Are you using relative path for load XML file?
If yes, change it to full path. Get you application path and add your file name.
